### PR TITLE
DevHub publishing pull request

### PR DESCRIPTION
**Pull request by TaylorFries**
## 🎯 Summary

  
[MVP-180](https://citz-imb.atlassian.net/browse/MVP-180)

This pull request will add the files required to publish documentation on DevHub

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

**Review by SDToews — DISMISSED** [review]

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Build TechDocs with DevHub TechDocs Publish Action
```
Comment (BradyMitch) [inline .github/workflows/publish.yaml:1]: Id rename this file to publish-techdocs.yaml

File: .github/workflows/publish-techdocs.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Build TechDocs with DevHub TechDocs Publish Action
+
+on:
+    workflow_dispatch:
+    release:
```
Comment (TaylorFries) [inline .github/workflows/publish-techdocs.yaml:5]: not sure that this will work.. but we can try?

File: techdocs/docs/authentication-flow.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Authentication Flow
+
+Below is a flow diagram of how authentication works between the SSO packages. 
+
+![alt text](../../assets/sso_auth_flow.PNG)
```
Comment (TaylorFries) [inline techdocs/docs/authentication-flow.md:5]: not sure if this path will work till pushing to dev site

File: techdocs/docs/using-the-package/custom-session-expiery-function.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <SSOProvider onRefreshExpiry={() => customTokenExpiry()}>
+        <App />
+    </SSOProvider>
+  </React.StrictMode>
+);
+```
+
+If you are building your own dialog, you can reference how the default dialog is made from the files below:  
+[Provider](https://github.com/bcgov/citz-imb-sso-react/blob/main/src/components/Provider.tsx)  
+[RefreshExpiryDialog](https://github.com/bcgov/citz-imb-sso-react/blob/main/src/components/RefreshExpiryDialog)
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/custom-session-expiery-function.md:31]: Do we think these will work?

Comment (BradyMitch) [reply techdocs/docs/using-the-package/custom-session-expiery-function.md:31]: I dont see why they wouldnt work

File: techdocs/docs/authentication-flow.md (hunk 3 above)
Comment (BradyMitch) [reply techdocs/docs/authentication-flow.md:5]: Might need to link to the image url but we can try this

File: .github/workflows/publish-techdocs.yaml (hunk 2 above)
Comment (BradyMitch) [inline .github/workflows/publish-techdocs.yaml:5]:
```suggestion
    release:
      types: [released]
```

**Comment by TaylorFries** [comment]
> looks like some of the docs files are using uppercase and some lowercase, can we make them consistent?

changed all files to use kebab-case

File: .github/workflows/publish-techdocs.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      types: [released]
+        branches: [ main ]
+        paths:
```
Comment (BradyMitch) [inline .github/workflows/publish-techdocs.yaml:8]: Paths need to be updated to reflect the change from `docs` to `techdocs/docs`

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**